Switch from absolute to flex/relative styles and strip unused API and features

fahrradflucht opened the pull request.

fahrradflucht committed:
Add basic test cases

fahrradflucht committed:
Remove vertical feature

fahrradflucht committed:
Remove range feature

fahrradflucht committed:
Remove allowCross and pushalbe fetures

This features are range related and therefore obsolete.

fahrradflucht committed:
Remove docs for removed API

fahrradflucht committed:
Remove array prop types from values (range related)

fahrradflucht committed:
Remove vertical styles

fahrradflucht committed:
switch to flex styles for marks

fahrradflucht committed:
switch to flex styles for steps

fahrradflucht committed:
restructure vertical positioning

fahrradflucht committed:
Remove tooltip feature

fahrradflucht committed:
Remove obsolete examples

fahrradflucht committed:
Fix lint errors

fahrradflucht committed:
Move handle inside the scale component

fahrradflucht committed:
Fix linter setup and errors

fahrradflucht committed:
Delete obsolete tests

fahrradflucht committed:
Remove unused babel-runtime dependency

fahrradflucht committed:
Reset version number

unstubbable committed:
add .npmignore file

unstubbable committed:
0.0.2

fahrradflucht committed:
Calculate vertical styles inline

The vertical alignment of dots and marks broke on mark counts other then 3 before. This should work now with n dot/mark counts.

unstubbable committed:
1.0.0

fahrradflucht committed:
Update README

fahrradflucht committed:
Add prepublish script

fahrradflucht committed:
1.0.1

fahrradflucht committed:
6.0.0

unstubbable commented on README.md:
```diff
@@ -200,18 +134,8 @@ npm start
-## Coverage
-
-http://localhost:8005/node_modules/rc-server/node_modules/node-jscover/lib/front-end/jscoverage.html?w=http://localhost:8088/tests/runner.html?coverage
+http://localhost:8005/examples/basics.js
```
=> http://localhost:8005/examples/basics.html

unstubbable commented on src/Handle.jsx:
```diff
@@ -1,81 +1,24 @@
-        onMouseEnter={this.showTooltip.bind(this)}
-        onMouseLeave={this.hideTooltip.bind(this)}
-      />
+      <div className={className} style={style} />
```
return here

unstubbable commented on src/Scale.jsx:
```diff
@@ -0,0 +1,109 @@
+const createMarks = (className, marks, included, upperBound, lowerBound) => {
+  const marksKeys = Object.keys(marks);
+
+  const elements = marksKeys.map(parseFloat).sort((a, b) => a - b).map((point) => {
```
return here

unstubbable commented on assets/index.less:
```diff
@@ -23,37 +22,36 @@
     border-radius: 50%;
     border: solid 2px tint(@primary-color, 50%);
     background-color: #fff;
+    z-index: 1000;
```
Isn't `2` enough?

unstubbable commented on examples/basics.js:
```diff
@@ -0,0 +1,62 @@
+  100: 100,
+};
+
+const fourMarks = {
```
Can you use strings as marks for one of the examples?

unstubbable commented on src/Scale.jsx:
```diff
@@ -0,0 +1,109 @@
+
+const Scale = ({ prefixCls, marks, dots, step, included, handles,
+                tracks, lowerBound, upperBound, max, min }) => {
+  const dotNodes = calcPoints(marks, dots, step, min, max).map((point) => {
```
Sometimes you call them "nodes", sometimes "elements". I would always use "elements".

unstubbable commented on src/Scale.jsx:
```diff
@@ -0,0 +1,109 @@
+const Scale = ({ prefixCls, marks, dots, step, included, handles,
+                tracks, lowerBound, upperBound, max, min }) => {
+  const dotNodes = calcPoints(marks, dots, step, min, max).map((point) => {
+    const isActived = (!included && point === upperBound) ||
```
`isActive`?

fahrradflucht committed:
Set handle z-index to a reasonable value

It is still 10 although 2 would be sufficient to leave room
for children styling.

fahrradflucht committed:
Change marks on one example to strings

fahrradflucht committed:
Refactor Handle to a stateless functional component

fahrradflucht committed:
Use consistent naming for scale elements

fahrradflucht committed:
Fix example URL in README
